### PR TITLE
Use Firestore V1 in Google.Cloud.Firestore

### PR DESCRIPTION
**Pull request by jskeet**
All the unit tests, integration tests and snippets now pass.

### CODE REVIEW

File: apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                ThrowInvalidResource();
+            }
+            int startOfDatabaseId = endOfProjectId + 1 + databasesPrefix.Length;
+            if (startOfDatabaseId == resource.Length)
+            {
+                // No database ID
+                ThrowInvalidResource();
+            }
+            int endOfDatabaseId = resource.IndexOf('/', startOfDatabaseId);
+            // It's valid for the whole resource name to be the database name
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs]: nit: I find this comment confusing here. Should it be on top of the return statement instead?

File: apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        
+        // Visible for testing
+
+        /// <summary>
+        /// Retrieves the database resource name from a full resource name.
+        /// Validation is performed as far as the database ID but no further; the database ID is deemed to end at the first slash.
+        /// </summary>
+        /// <param name="resource">The resource name, which must start with projects/{project_id}/databases/{database_id}</param>
+        /// <returns>The database resource name</returns>
+        internal static string GetDatabaseResourceName(string resource)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs]: Maybe add a couple of unit tests? Ignore if those are coming in the next commits. And also fine with that being done later.

File: apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs]: Moved.

File: apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.Firestore.V1/Google.Cloud.Firestore.V1/FirestoreClientPartial.cs]: As it happens, there's a test file I'd forgotten to copy over from V1Beta1 :) Will do that now.

**Comment by jskeet** [comment]
I've copied the test, changed the comment and rebased so that it's in the same commit that introduces the resource routing.

**Pull request merged**